keanu-project/src/main/java/io/improbable/keanu/util/csv/CsvWriter.java

```diff
@@ -0,0 +1,100 @@
+package io.improbable.keanu.util.csv;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class CsvWriter {
+
+    private static final String DEFAULT_DELIMITER = ",";
+    private static final String DEFAULT_SUFFIX = ".csv";
+    private static final String NEWLINE = "\n";
+
+    private final List<List<String>> data;
+    private String delimiter;
+    private boolean writeHeader;
+    private List<String> header;
+    private FileWriter fileWriter;
+
+    public CsvWriter(List<List<String>> data) {
+        this.data = data;
+        this.delimiter = DEFAULT_DELIMITER;
+        this.writeHeader = false;
+        this.header = null;
+        this.fileWriter = null;
+    }
+
+    public File toFile(String filename, String directory) {
+        return toFile(filename, new File(directory));
+    }
+
+    public File toFile(String filename, File directory) {
+        try {
+            return toFile(File.createTempFile(filename, DEFAULT_SUFFIX, directory));
```

Comment:
using `File.createTempFile(...)` generates a unique file name based off the args. I think you want `toFile(directory.toPath().resolve(filename).toFile())` and since there's no file created here, you can just return that instead of try catching it.

Reply:
awesome, changed